src/System.Private.CoreLib/shared/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -13,10 +13,10 @@ public abstract partial class EqualityComparer<T> : IEqualityComparer, IEquality
     {
         // public static EqualityComparer<T> Default is runtime-specific
 
-        public abstract bool Equals(T x, T y);
-        public abstract int GetHashCode(T obj); // TODO-NULLABLE-GENERIC: Shouldn't accept nulls.
+        public abstract bool Equals([AllowNull] T x, [AllowNull] T y);
+        public abstract int GetHashCode([DisallowNull] T obj);
 
-#pragma warning disable CS8617 // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/30958
+#pragma warning disable CS8617 // TODO-NULLABLE: Covariant parameter types (https://github.com/dotnet/roslyn/issues/30958)
```

Comment:
Should we instead move this NoWarn to the csproj?

Reply:
I considered that but decided against it (and actually started to remove the existing CS8609 that's NoWarn but undid it to not add a whole bunch of additional work to this PR).  We actually want the warnings to fire when they're relevant.